pkg/clients/hostedzone/hostedzone.go

```diff
@@ -68,7 +68,9 @@ func LateInitialize(spec *v1alpha1.HostedZoneParameters, obs *route53.GetHostedZ
 	if obs == nil || obs.HostedZone == nil {
 		return
 	}
-	spec.DelegationSetID = awsclients.LateInitializeStringPtr(spec.DelegationSetID, obs.DelegationSet.Id)
+	if spec.DelegationSetID == nil && obs.DelegationSet.Id != nil {
```

Comment:
Did you manage to reproduce the panic? I suspect `obs.DelegationSet` is the nil-check we need, as @poblish mentions.

* `spec.DelegationSetID == nil` is checked by `LateInitializeStringPtr`
* `obs.DelegationSet.Id == nil` would be a noop, we would only assign that `nil` to `spec.DelegationSetID` in the case where it is already `nil`, due to the guard above.

Either way, unless `obs.DelegationSet` is `nil` I fail to see how this could panic.

So I wonder if the correct fix is:

```suggestion
	if obs.DelegationSet != nil {
```

Reply:
yes - reproduce with private hosted dns zone - i checked your suggestion lgtm i changed my pr
thanks for review